Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,358 @@
+    public func authenticate (request: RouterRequest,
+                              response: RouterResponse,
+                              options: [String:Any],
+                              onSuccess: @escaping (UserProfile) -> Void,
+                              onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                              onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                              inProgress: @escaping () -> Void) {
+        // If Step 2, is accomplished, the query will have `oauth_verifier` and `oauth_token`. The oauth_token should
+        // match the OAuthToken you got during Step 1.
+        if let verifier = request.queryParameters["oauth_verifier"],
+            let newOAuthToken = request.queryParameters["oauth_token"],
+            newOAuthToken == oAuthToken {
+            oAuthTokenVerifier = verifier
+            twitterAccessTokenRequest(request: request,
+                                      response: response,
+                                      onSuccess: onSuccess,
+                                      onFailure: onFailure,
+                                      onPass: onPass)
+        }
+        else { // Start with Step 1.
+            twitterTokenRequest(request: request,
+                                response: response,
+                                onFailure: onFailure,
+                                onPass: onPass,
+                                inProgress: inProgress)
+        }
+    }
+
+    //MARK: Step 1: Obtaining a request token
+    
+    /// The method gathers the necessary components to create a request to https://api.twitter.com/oauth/request_token. 
+    /// If the request is successful, it will move on to Step 2 which redirects the user. Failure or incorrect payloads
+    /// call `onFailure` or `onPass`, respectively.
+    ///
+    /// - Parameters:
+    ///   - request: The `RouterRequest` object used to get information about the request.
+    ///   - response: The `RouterResponse` object used to respond to the request.
+    ///   - onFailure: The closure to invoke in the case of an authentication failure.
+    ///   - onPass: The closure to invoke when the plugin doesn't recognize the authentication data (usually an
+    ///       authentication token) in the request.
+    ///   - inProgress: The closure to invoke to cause a redirect to the login page in the case of redirecting
+    ///       authentication.
+    func twitterTokenRequest(request: RouterRequest,
+                             response: RouterResponse,
+                             onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                             onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                             inProgress: @escaping () -> Void) {
+        var requestOptions: [ClientRequest.Options] = []
+        requestOptions.append(.schema("https://"))
```

Comment:
I was following FB but I thought it was for legibility. Turns out, it's just as legible.